### ISSUE TITLE
settings for multiple vector tile sources

### ISSUE THREAD

**Issue by musicformellons** [mapzen/tileserver#44]
I am looking at how to combine multiple vector tile files. Could you explain what settings are needed to have a single entry point containing vector tile data of both (or multiple) sources?

**Comment by nvkelso** [comment]
You mean like the "all" alias that the Mapzen be Tor tile service uses? That's controlled in the main config.yaml file in vector-datasource.

**Comment by musicformellons** [comment]
I am not sure I understand, so I will try to be more specific. I would like to use Mapzen vector-datasource (mvt) tiles and combine these with an extra custom layer in vector tile (mvt) format which consists of points (actually a large geojson file converted into mbtiles). I use mapbox-gl-js and there I find (this is mapbox-gl but at mapbox-gl-js they reference to this, so it applies to both):
https://github.com/mapbox/mapbox-gl-native/issues/3141#issuecomment-160581589 
tl;dr: Should be solved on server side (and thus both sources should come from same server). So I should have my own vector-data tiles and my custom layer and solve it at the server. I am not exactly sure how to do that:
- Do you first have 'to mingle' (compositing) them together at database level?
- Or could just serve multiple tiles and set the server to do so?

Hope it makes sense.

**Comment by musicformellons** [comment]
Could it be you mean this 'all' in vector-datasource/**queries**.yaml ?

all:
- water
- earth
- places
- landuse
- roads
- buildings
- pois
- boundaries
- transit

**Comment by musicformellons** [comment]
These guys make it look easy:
https://github.com/osm2vectortiles/osm2vectortiles/issues/26

stating: "vector tiles are designed in a way which supports merging by direct binary concatenation."

**Comment by nvkelso** [comment]
Oops, I had a weird auto correct about Tor in my last comment! I meant to say Mapzen vector tiles.

We use tilezen/tileserver for decomposition:

https://github.com/tilezen/tileserver/blob/master/tileserver/__init__.py

It listens for requests to for certain layers - if a layer other than "all" is asked for them we'll peal out just those layers to return in the response (which is comparable with osm2vectortiles statement). 

But this only decomposes the general Tilezen tiles, it doesn't (yet) have logic to combine &/or decompose any arbitrary tile data from arbitrary sources.

**Comment by musicformellons** [comment]
Thanks. That would be a useful feature to add I think.

**Comment by nvkelso** [comment]
Closing for inactivity.
